app/src/main/java/com/example/poste/activities/EditPostActivity.java

```diff
@@ -59,84 +39,73 @@ public class EditPostActivity extends AppCompatActivity {
-
-        // Cancel button click handler
-        cancelBtn.setOnClickListener(view -> {
-            // Send back to folder view
-            sendToDashboard();
+        buttonCancelChanges.setOnClickListener(view -> {
+            Intent returnToFolderIntent = new Intent(EditPostActivity.this, EditPostActivity.class);
+            finish();
+            returnToFolderIntent.putExtra("postID",postId);
+            returnToFolderIntent.putExtra("folderID",folderId);
+            startActivity(returnToFolderIntent);
         });
-
-        // Save button click handler
-        saveBtn.setOnClickListener(view -> {
-            try {
-                String selectedFolderName = spinner.getSelectedItem().toString();
-                Integer selectedFolderId = folderIdNameMap.get(selectedFolderName);
-
-                // Save the post in the API
-                if (currentPost.getName() != postNameField.getText().toString() ||
-                        currentPost.getLink() != postLinkField.getText().toString()) {
-                    API.updatePost(currentPost.getId(), postNameField.getText().toString(), postLinkField.getText().toString(), currentPost.getOwnerId());
-                }
-
-                // Update the post in the API
-                if (currentFolder.getId() != selectedFolderId) {
-                    API.removePostFromFolder(currentPost.getId(), currentFolder.getId());
-                    API.addPostToFolder(currentPost.getId(), selectedFolderId);
-                }
-
-                // Send back to folder view
-                sendToDashboard();
-            } catch (APIException e) {
-                throw new RuntimeException(e);
+        buttonSaveChanges.setOnClickListener(view -> {
+            if (currentPost!= null){
+                String title = postTitle.getText().toString();
+                String description = postDescription.getText().toString();
+                String url = linkToPost.getText().toString();
+
+                MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+                Call<ResponseBody> call = apiService.editPost(User.getUser().getTokenHeader(),
+                        postId,
+                        new EditPostRequest(title,description,url));
+
+                call.enqueue(new Callback<ResponseBody>() {
+                    @Override
+                    public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                        if (response.isSuccessful()){
+                            currentPost.setTitle(title);
+                            currentPost.setDescription(description);
+                            currentPost.setUrl(url);
+                            Toast.makeText(EditPostActivity.this,"Edit post successful!", Toast.LENGTH_SHORT).show();
+                        } else {
+                            Toast.makeText(EditPostActivity.this,"Post to edit not found!", Toast.LENGTH_SHORT).show();
```

Comment:
Might be beneficial to revisit this and pull the error that the server is returning to the client; while 'not found' is one possible cause, there could be others (token expired, not authorized or no access, etc)

Reply:
We should be able to parse out the error message from the server response directly and show that (or use it to determine what toast to make, at the least) -- I only see "Post to edit not found" or "Edit failed due to server error". 

Incidentally, the onFailure means a _request_ error, which likely indicates a connectivity error (means no reply was received from the server at all)